### PR TITLE
Feature/790/113 executor instrument

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

- Changed loop from an endless busy loop to a task that reenqueues itself
- Use scheduled executor instead of thread sleep

## Why it does that

- Reduce thread usage
- Running one executor task per loop execution will allow metrics on loop execution time (next PR)

## Linked Issue(s)

Relates to #113

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Comment by cpeeyush** [comment]
Do we also want to add instrumentation for executor service in DPF `extensions/data-plane/data-plane-spi/src/main/java/org/eclipse/dataspaceconnector/dataplane/spi/pipeline/ParallelSink.java` ?

File: common/state-machine-lib/src/main/java/org/eclipse/dataspaceconnector/common/statemachine/StateMachine.java [hunk 1]
```diff
@@ -94,42 +101,46 @@ public boolean isActive() {
...
-                        monitor.severe(format("StateMachine [%s] error caught", name), e);
-                        Thread.sleep(waitStrategy.retryInMillis());
-                    } catch (InterruptedException ex) {
-                        active.set(false);
-                        monitor.severe(format("StateMachine [%s] unrecoverable error", name), e);
-                    }
                 }
             }
+            if (active.get()) {
+                submit();
```
[the remark this answers is not in the document]
Comment (algattik) [reply common/state-machine-lib/src/main/java/org/eclipse/dataspaceconnector/common/statemachine/StateMachine.java]: Added comment

File: spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/system/NullExecutorInstrumentation.java [hunk 2]
```diff
@@ -12,12 +12,10 @@
...
-package org.eclipse.dataspaceconnector.core.executor;
-
-import org.eclipse.dataspaceconnector.spi.system.ExecutorInstrumentation;
+package org.eclipse.dataspaceconnector.spi.system;
 
 /**
  * Default implementation of a {@link ExecutorInstrumentation} that performs no action.
  */
-public class NoopExecutorInstrumentation implements ExecutorInstrumentation {
+public class NullExecutorInstrumentation implements ExecutorInstrumentation {
```
[the remark this answers is not in the document]
Comment (algattik) [reply spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/system/NullExecutorInstrumentation.java]: Reduced scope of this PR so this change is now not in scope

**Review by cpeeyush — APPROVED** [review]
Approving. Discussed offline some of the open items will be addressed in a separate follow up PR.

File: common/state-machine-lib/src/main/java/org/eclipse/dataspaceconnector/common/statemachine/StateMachine.java [hunk 3]
```diff
@@ -57,11 +59,17 @@ private StateMachine(String name, Monitor monitor, WaitStrategy waitStrategy) {
...
      */
     public Future<?> start() {
         active.set(true);
-        return executor.submit(loop());
+        return submit(0L);
+    }
+
+    @NotNull
+    private Future<?> submit(long delayMillis) {
+        monitor.severe(format("StateMachine %s delaying for %s ms", name, delayMillis));
```
Comment (cpeeyush) [inline common/state-machine-lib/src/main/java/org/eclipse/dataspaceconnector/common/statemachine/StateMachine.java:71]: nit:
```suggestion
        monitor.severe(format("StateMachine %s delaying for %d ms", name, delayMillis));
```

**Review by ouphi — APPROVED** [review]

**Pull request merged**